odl/discr/discretization.py

```diff
@@ -241,40 +225,94 @@ def domain(self):
         """Domain of the continuous space."""
         return self.uspace.domain
 
+    # Pass-through attributes of the wrapped ``dspace``
+    def zero(self):
+        """Return the element of all zeros."""
+        return self.element_type(self, self.dspace.zero())
+
+    def one(self):
+        """Return the element of all ones."""
+        return self.element_type(self, self.dspace.one())
+
+    @property
+    def weighting(self):
+        """This space's weighting scheme."""
+        return self.dspace.weighting
+
+    @property
+    def is_weighted(self):
+        """``True`` if the ``dspace`` is weighted."""
+        return getattr(self.dspace, 'is_weighted', False)
+
+    def _lincomb(self, a, x1, b, x2, out):
+        """Raw linear combination."""
+        self.dspace._lincomb(a, x1.tensor, b, x2.tensor, out.tensor)
+
+    def _dist(self, x1, x2):
+        """Raw distance between two elements."""
+        return self.dspace._dist(x1.tensor, x2.tensor)
+
+    def _norm(self, x):
+        """Raw norm of an element."""
+        return self.dspace._norm(x.tensor)
+
+    def _inner(self, x1, x2):
+        """Raw inner product of two elements."""
+        return self.dspace._inner(x1.tensor, x2.tensor)
+
+    def _multiply(self, x1, x2, out):
+        """Raw pointwise multiplication of two elements."""
+        self.dspace._multiply(x1.tensor, x2.tensor, out.tensor)
+
+    def _divide(self, x1, x2, out):
+        """Raw pointwise multiplication of two elements."""
+        self.dspace._divide(x1.tensor, x2.tensor, out.tensor)
+
+    @property
+    def examples(self):
+        """Return example functions in the space.
+
+        These are created by discretizing the examples in the underlying
+        `uspace`.
+
+        See Also
+        --------
+        odl.space.fspace.FunctionSpace.examples
+        """
+        for name, elem in self.uspace.examples:
+            yield (name, self.element(elem))
+
     @property
     def element_type(self):
-        """`DiscretizedSetElement`"""
-        return DiscretizedSetElement
+        """Type of elements in this space: `DiscretizedSpaceElement`."""
+        return DiscretizedSpaceElement
 
 
-class DiscretizedSetElement(NtuplesBaseVector):
+class DiscretizedSpaceElement(Tensor):
 
-    """Representation of a `DiscretizedSet` element.
+    """Representation of a `DiscretizedSpace` element.
 
     Basically only a wrapper class for dspace's element class."""
 
-    def __init__(self, space, ntuple):
+    def __init__(self, space, tensor):
         """Initialize a new instance."""
-        assert isinstance(space, DiscretizedSet)
-        assert ntuple in space.dspace
-
-        NtuplesBaseVector.__init__(self, space)
-        self.__ntuple = ntuple
+        super().__init__(space)
```

Comment:
Oh, I actually didn't want to change that one.

But anyway, the non-speed-critical ones should use `super`, simply because it allows dependency injection, something that can be very useful when subclassing. An we should change the speed-critical ones also as soon as Python 2 is dropped.